File: lib/screens/ayuda_screen.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import '../utils/colors.dart';

class AyudaScreen extends StatelessWidget {
  const AyudaScreen({super.key});

  final List<_Pregunta> _preguntas = const [
    _Pregunta(
      '¿Cómo me registro?',
      'Tocá "Registrate" en la pantalla de login, completá tus datos y elegí tu club. Necesitás un email válido y una contraseña de al menos 6 caracteres.',
    ),
    _Pregunta(
      '¿Cómo inicio con huella?',
      'Primero registrate con email. Después, cada vez que quieras entrar, podés usar tu huella dactilar si tu dispositivo lo soporta.',
    ),
    _Pregunta(
      '¿Puedo cambiar de club?',
      'Por ahora no. Una vez que elegiste tu club al registrarte, queda fijo. En futuras versiones se podrá editar.',
    ),
    _Pregunta(
      '¿Los datos son reales?',
      'Los datos de torneos, partidos, posiciones y goleadores son de ejemplo para demostración. En producción se conectarían a la base de datos oficial de la AHBA.',
    ),
    _Pregunta(
      '¿Cómo funcionan los filtros?',
      'En cada pantalla hay chips arriba que te permiten filtrar por categoría, división, estado, etc. Tocá el que quieras y la lista se actualiza.',
    ),
    _Pregunta(
      '¿Qué pasa si no aparece mi club?',
      'Tenemos cargados los 140 clubes principales de la AHBA. Si el tuyo no aparece, contactanos para agregarlo.',
    ),
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.background,
      appBar: AppBar(
        title: Text('Soporte', style: GoogleFonts.montserrat(fontWeight: FontWeight.bold)),
        backgroundColor: AppColors.primary,
        foregroundColor: Colors.white,
        elevation: 0,
      ),
      body: SingleChildScrollView(
        child: Column(
          children: [
            // Header
            Container(
              padding: const EdgeInsets.fromLTRB(20, 10, 20, 24),
              decoration: const BoxDecoration(
                color: AppColors.primary,
                borderRadius: BorderRadius.only(
                  bottomLeft: Radius.circular(24),
                  bottomRight: Radius.circular(24),
                ),
              ),
              child: Row(
                children: [
                  Container(
                    padding: const EdgeInsets.all(12),
                    decoration: BoxDecoration(color: Colors.white.withOpacity(0.1), borderRadius: BorderRadius.circular(16)),
                    child: const Icon(Icons.help_center_rounded, color: AppColors.secondary, size: 28),
                  ),
                  const SizedBox(width: 16),
                  const Expanded(
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text('¿Cómo podemos ayudarte?', style: TextStyle(color: Colors.white, fontSize: 16, fontWeight: FontWeight.bold)),
                        SizedBox(height: 2),
                        Text('Preguntas frecuentes y soporte técnico', style: TextStyle(color: Colors.white70, fontSize: 11)),
                      ],
                    ),
                  ),
                ],
              ),
            ),

            Padding(
              padding: const EdgeInsets.all(20),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  _buildSectionTitle('PREGUNTAS FRECUENTES'),
                  const SizedBox(height: 16),
                  ..._preguntas.asMap().entries.map((entry) => _buildPreguntaCard(entry.key, entry.value)),
                  
                  const SizedBox(height: 30),
                  
                  _buildSectionTitle('CONTACTO DIRECTO'),
                  const SizedBox(height: 16),
                  _buildContactCard(),
                  const SizedBox(height: 30),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildSectionTitle(String title) {
    return Row(
      children: [
        Container(width: 4, height: 16, decoration: BoxDecoration(color: AppColors.secondary, borderRadius: BorderRadius.circular(2))),
        const SizedBox(width: 10),
        Text(title, style: GoogleFonts.montserrat(fontSize: 12, fontWeight: FontWeight.w800, color: AppColors.textPrimary, letterSpacing: 1)),
      ],
    );
  }

  Widget _buildPreguntaCard(int index, _Pregunta p) {
    return Container(
      margin: const EdgeInsets.only(bottom: 12),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(16),
        boxShadow: [
          BoxShadow(color: Colors.black.withOpacity(0.02), blurRadius: 10, offset: const Offset(0, 4))
        ],
      ),
      child: Theme(
        data: ThemeData().copyWith(dividerColor: Colors.transparent),
        child: ExpansionTile(
          iconColor: AppColors.primary,
          collapsedIconColor: Colors.grey,
          leading: CircleAvatar(
            backgroundColor: AppColors.primary.withOpacity(0.1),
            radius: 14,
            child: Text('${index + 1}', style: const TextStyle(color: AppColors.primary, fontSize: 12, fontWeight: FontWeight.bold)),
          ),
          title: Text(p.pregunta, style: GoogleFonts.montserrat(fontWeight: FontWeight.bold, fontSize: 13, color: AppColors.textPrimary)),
          children: [
            Padding(
              padding: const EdgeInsets.fromLTRB(20, 0, 20, 20),
              child: Text(p.respuesta, style: const TextStyle(color: AppColors.textSecondary, fontSize: 13, height: 1.5)),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildContactCard() {
    return Container(
      padding: const EdgeInsets.all(24),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(24),
        boxShadow: [
          BoxShadow(color: Colors.black.withOpacity(0.02), blurRadius: 10, offset: const Offset(0, 4))
        ],
      ),
      child: Column(
        children: [
          const Icon(Icons.alternate_email_rounded, color: AppColors.primary, size: 40),
          const SizedBox(height: 16),
          Text('¿Tenés otra consulta?', style: GoogleFonts.montserrat(fontSize: 16, fontWeight: FontWeight.bold)),
          const SizedBox(height: 8),
          const Text('Escribinos y te responderemos a la brevedad.', textAlign: TextAlign.center, style: TextStyle(color: AppColors.textSecondary, fontSize: 12)),
          const SizedBox(height: 20),
          Container(
            padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 12),
            decoration: BoxDecoration(color: AppColors.primary.withOpacity(0.05), borderRadius: BorderRadius.circular(12)),
            child: const Text('soporte@hockeyahba.com.ar', style: TextStyle(color: AppColors.primary, fontWeight: FontWeight.bold, fontSize: 14)),
          ),
        ],
      ),
    );
  }
}

class _Pregunta {
  final String pregunta;
  final String respuesta;

  const _Pregunta(this.pregunta, this.respuesta);
}

File: app/src/main/java/com/example/hockey_app/ui/screens/profile/AyudaScreen.kt
package com.example.hockey_app.ui.screens.profile

import android.content.Intent
import android.net.Uri
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

private data class Pregunta(val pregunta: String, val respuesta: String)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AyudaScreen(
    onBack: () -> Unit
) {
    val preguntas = listOf(
        Pregunta(
            "¿Cómo me registro?",
            "Tocá \"Registrate\" en la pantalla de login, completá tus datos y elegí tu club. Necesitás un email válido y una contraseña de al menos 6 caracteres."
        ),
        Pregunta(
            "¿Cómo inicio con huella?",
            "Primero registrate con email. Después, cada vez que quieras entrar, podés usar tu huella dactilar si tu dispositivo lo soporta."
        ),
        Pregunta(
            "¿Puedo cambiar de club?",
            "Por ahora no. Una vez que elegiste tu club al registrarte, queda fijo. En futuras versiones se podrá editar."
        ),
        Pregunta(
            "¿Los datos son reales?",
            "Los datos de torneos, partidos, posiciones y goleadores son de ejemplo para demostración. En producción se conectarían a la base de datos oficial de la AHBA."
        ),
        Pregunta(
            "¿Cómo funcionan los filtros?",
            "En cada pantalla hay chips arriba que te permiten filtrar por categoría, división, estado, etc. Tocá el que quieras y la lista se actualiza."
        ),
        Pregunta(
            "¿Qué pasa si no aparece mi club?",
            "Tenemos cargados los 140 clubes principales de la AHBA. Si el tuyo no aparece, contactanos para agregarlo."
        ),
    )
    val context = LocalContext.current

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Soporte", fontWeight = FontWeight.Bold) },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Volver")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.primary,
                    titleContentColor = Color.White,
                    navigationIconContentColor = Color.White
                )
            )
        },
        containerColor = MaterialTheme.colorScheme.background
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .verticalScroll(rememberScrollState())
        ) {
            // Header Banner
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(bottomStart = 24.dp, bottomEnd = 24.dp))
                    .background(MaterialTheme.colorScheme.primary)
                    .padding(horizontal = 20.dp, vertical = 16.dp)
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Box(
                        modifier = Modifier
                            .size(50.dp)
                            .clip(RoundedCornerShape(16.dp))
                            .background(Color.White.copy(alpha = 0.1f)),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(Icons.Default.HelpCenter, contentDescription = null, tint = MaterialTheme.colorScheme.secondary, modifier = Modifier.size(28.dp))
                    }
                    Spacer(modifier = Modifier.width(16.dp))
                    Column {
                        Text("¿Cómo podemos ayudarte?", color = Color.White, fontWeight = FontWeight.Bold, fontSize = 16.sp)
                        Text("Preguntas frecuentes y soporte técnico", color = Color.White.copy(alpha = 0.7f), fontSize = 11.sp)
                    }
                }
            }

            Column(modifier = Modifier.padding(20.dp)) {
                // FAQ Section
                SectionTitle("PREGUNTAS FRECUENTES")
                Spacer(modifier = Modifier.height(16.dp))
                preguntas.forEachIndexed { index, pregunta ->
                    FaqCard(index + 1, pregunta)
                    Spacer(modifier = Modifier.height(12.dp))
                }

                Spacer(modifier = Modifier.height(18.dp))

                // Contact Section
                SectionTitle("CONTACTO DIRECTO")
                Spacer(modifier = Modifier.height(16.dp))
                ContactCard(
                    onEmailClick = {
                        val intent = Intent(Intent.ACTION_SENDTO).apply {
                            data = Uri.parse("mailto:soporte@hockeyahba.com.ar")
                        }
                        context.startActivity(intent)
                    }
                )
                Spacer(modifier = Modifier.height(30.dp))
            }
        }
    }
}

@Composable
private fun SectionTitle(title: String) {
    Row(verticalAlignment = Alignment.CenterVertically) {
        Box(modifier = Modifier.width(4.dp).height(16.dp).clip(CircleShape).background(MaterialTheme.colorScheme.secondary))
        Spacer(modifier = Modifier.width(10.dp))
        Text(title, fontWeight = FontWeight.ExtraBold, fontSize = 12.sp, letterSpacing = 1.sp)
    }
}

@Composable
private fun FaqCard(index: Int, pregunta: Pregunta) {
    var expanded by remember { mutableStateOf(false) }
    Card(
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = Color.White),
        elevation = CardDefaults.cardElevation(1.dp)
    ) {
        Column {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Box(
                    modifier = Modifier.size(28.dp).clip(CircleShape).background(MaterialTheme.colorScheme.primary.copy(alpha = 0.1f)),
                    contentAlignment = Alignment.Center
                ) {
                    Text("$index", color = MaterialTheme.colorScheme.primary, fontSize = 12.sp, fontWeight = FontWeight.Bold)
                }
                Spacer(modifier = Modifier.width(12.dp))
                Text(pregunta.pregunta, fontWeight = FontWeight.Bold, fontSize = 13.sp, modifier = Modifier.weight(1f))
                IconButton(onClick = { expanded = !expanded }) {
                    Icon(
                        if (expanded) Icons.Default.ExpandLess else Icons.Default.ExpandMore,
                        contentDescription = null,
                        tint = if (expanded) MaterialTheme.colorScheme.primary else Color.Gray
                    )
                }
            }
            if (expanded) {
                Text(
                    pregunta.respuesta,
                    modifier = Modifier.padding(start = 56.dp, end = 20.dp, bottom = 20.dp),
                    fontSize = 13.sp,
                    color = Color.Gray,
                    lineHeight = 20.sp
                )
            }
        }
    }
}

@Composable
private fun ContactCard(onEmailClick: () -> Unit) {
    Card(
        shape = RoundedCornerShape(24.dp),
        colors = CardDefaults.cardColors(containerColor = Color.White),
        elevation = CardDefaults.cardElevation(1.dp)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(24.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Icon(Icons.Default.AlternateEmail, contentDescription = null, tint = MaterialTheme.colorScheme.primary, modifier = Modifier.size(40.dp))
            Spacer(modifier = Modifier.height(16.dp))
            Text("¿Tenés otra consulta?", fontWeight = FontWeight.Bold, fontSize = 16.sp)
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                "Escribinos y te responderemos a la brevedad.",
                textAlign = androidx.compose.ui.text.style.TextAlign.Center,
                color = Color.Gray,
                fontSize = 12.sp
            )
            Spacer(modifier = Modifier.height(20.dp))
            OutlinedButton(
                onClick = onEmailClick,
                shape = RoundedCornerShape(12.dp)
            ) {
                Icon(Icons.Default.Email, contentDescription = null, modifier = Modifier.size(16.dp))
                Spacer(modifier = Modifier.width(8.dp))
                Text("soporte@hockeyahba.com.ar", fontWeight = FontWeight.Bold, fontSize = 13.sp)
            }
        }
    }
}
